include/mata/util.hh

```diff
@@ -419,6 +420,100 @@ struct TuplePrinter<Tuple, 1> {
     }
 };
 
+// This reserves space in a vector, to be used before push_back or insert.
+// Assuming the doubling extension strategy, it only makes the first reserve large, after that it leaves it to the doubling.
+// Might be worth thinking about it.
+//  around 30% speedup for fragile revert,
+//  more than 50% for simple revert,
+// (when testing on a stupid test case)
+template<class Vector>
+void inline reserve_on_insert(Vector & vec,size_t needed_capacity = 0,size_t extension = 64) {
```

Comment:
The `64` could be extracted to some constant/macro, that could be changed compile time maybe?